hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -102,6 +106,7 @@ func (r *NodePoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&hyperv1.NodePool{}).
 		// We want to reconcile when the HostedCluster IgnitionEndpoint is available.
 		Watches(&source.Kind{Type: &hyperv1.HostedCluster{}}, handler.EnqueueRequestsFromMapFunc(r.enqueueNodePoolsForHostedCluster)).
+		Watches(&source.Kind{Type: &hyperv1.HostedControlPlane{}}, handler.EnqueueRequestsFromMapFunc(hostedcluster.EnqueueParentHostedCluster)).
```

Comment:
If I'm reading correctly this will enqueue a hostedCluster reconcile.Request{} that will make the NodPool Reconcile function to return early since it'll try to find a NodePool with that NamespacedName and fail to find it? I see this as another symptom for https://github.com/openshift/hypershift/pull/1394/files#r875039237. I don't think this controller should be aware of HCP.

Reply:
Good catch, I will fix it up. Please read up in the other comment the rationale for doing it the way it is done before starting to interpret bugs.

Reply:
Fixed